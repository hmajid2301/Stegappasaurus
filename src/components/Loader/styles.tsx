import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  loaderContainer: {
    alignItems: "center",
    bottom: 0,
    flex: 1,
    justifyContent: "center",
    left: 0,
    opacity: 0.75,
    position: "absolute",
    right: 0,
    top: 0,
    zIndex: 2
  }
});

export default styles;
