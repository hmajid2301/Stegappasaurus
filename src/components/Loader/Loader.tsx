import * as React from "react";
import { ActivityIndicator, View } from "react-native";
import { colors } from "~/modules";

import styles from "./styles";

interface IProps {
  loading: boolean;
  overlay?: string;
}

const Loader = ({ loading, overlay }: IProps) => {
  if (loading) {
    return (
      <View
        style={[
          styles.loaderContainer,
          {
            backgroundColor: overlay
          }
        ]}
      >
        <ActivityIndicator color={colors.primary} size={"large"} />
      </View>
    );
  }

  return <View />;
};

export default Loader;
