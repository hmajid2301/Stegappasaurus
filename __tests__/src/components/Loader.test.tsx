import {shallow} from 'enzyme';
import React from 'react';

import Loader from '~/components/Loader';

describe('Loader: Match snapshots', () => {
  test('1', () => {
    const component = shallow(<Loader loading={true} />);
    expect(component).toMatchSnapshot();
  });

  test('2', () => {
    const component = shallow(<Loader loading={false} />);
    expect(component).toMatchSnapshot();
  });
});
