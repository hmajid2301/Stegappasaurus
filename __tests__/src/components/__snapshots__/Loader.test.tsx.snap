// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader: Match snapshots 1 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "bottom": 0,
        "flex": 1,
        "justifyContent": "center",
        "left": 0,
        "opacity": 0.75,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 2,
      },
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#009CFF"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`Loader: Match snapshots 2 1`] = `<View />`;
